src/components/Comments/WriteComment.js

```diff
@@ -118,27 +127,39 @@ const WriteComment = ({
             />
           </>
         )}
-
         <TextInput
           testID="changeinput"
           ref={commentInputRef}
           placeholder={StringConstant.commentBoxDefaultPlaceholder}
+          style={[styles.text, styles.content(isViewOnly)]}
           placeholderTextColor={COLORS.blackgrey}
-          style={[styles.text, styles.content]}
           onChangeText={onChangeText}
           value={value}
           multiline
           textAlignVertical="center"
+          pointerEvents={isViewOnly ? 'none' : 'auto'}
         />
-        <TouchableOpacity
-          testID="iscommentenable"
-          onPress={() => onPress(isAnonimity, anonimityData)}
-          style={styles.btn(!isCommentEnabled || loadingUser || loadingPost)}
-          disabled={!isCommentEnabled || loadingUser || loadingPost}>
-          <MemoSendComment
-            fillBackground={!isCommentEnabled || loadingUser ? COLORS.gray : COLORS.signed_primary}
-          />
-        </TouchableOpacity>
+        {!isViewOnly ? (
+          <TouchableOpacity
+            testID="iscommentenable"
+            onPress={() => onPress(isAnonimity, anonimityData)}
+            style={styles.btn(!isCommentEnabled || loadingUser || loadingPost)}
+            disabled={!isCommentEnabled || loadingUser || loadingPost}>
+            <MemoSendComment
+              fillBackground={
+                !isCommentEnabled || loadingUser ? COLORS.gray : COLORS.signed_primary
+              }
+            />
+          </TouchableOpacity>
+        ) : (
+          <View style={styles.isViewOnlyIcon}>
+            <MemoSendComment
+              fillBackground={
+                !isCommentEnabled || loadingUser ? COLORS.gray : COLORS.signed_primary
+              }
+            />
+          </View>
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The send comment button is now conditionally rendered based on the `isViewOnly` prop. This is a good way to prevent users from interacting with the component when it's in view-only mode.



<!-- This is an auto-generated comment by OSS CodeRabbit -->